api/include/mfxstructures.h

```diff
@@ -2269,6 +2269,48 @@ MFX_PACK_END()
 
 #endif
 
+#if (MFX_VERSION >= MFX_VERSION_NEXT)
+/* Multi-adapters Querying structs */
+typedef enum
+{
+    MFX_ENCODE_COMPONENT = 1,
+    MFX_DECODE_COMPONENT = 2,
+    MFX_VPP_COMPONENT    = 3
+} mfxComponentType;
+
+MFX_PACK_BEGIN_STRUCT_W_PTR()
+typedef struct
+{
+    mfxComponentType Type;
+    mfxVideoParam    Requirements;
+
+    mfxU16           reserved[4];
+} mfxComponentInfo;
+MFX_PACK_END()
+
+/* Adapter description */
+MFX_PACK_BEGIN_USUAL_STRUCT()
+typedef struct
+{
+    mfxPlatform Platform;
+    mfxU32      Number;
+
+    mfxU16      reserved[14];
```

Comment:
we might need adapter type on Linux to distinguish render nodes and cards

Reply:
Yep, those reserved fields might me reused for Linux support